buildSrc/src/test/groovy/edu/ucar/build/ui/ToolsUiJnlpExtensionTaskSpec.groovy

```diff
@@ -0,0 +1,133 @@
+package edu.ucar.build.ui
+
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.testfixtures.ProjectBuilder
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.gradle.testkit.runner.internal.PluginUnderTestMetadataReading
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import org.xmlunit.builder.DiffBuilder
+import org.xmlunit.builder.Input
+import org.xmlunit.diff.Diff
+import spock.lang.Specification
+
+/**
+ * Tests ToolsUiJnlpExtensionTask.
+ *
+ * @author cwardgar
+ * @since 2017-04-05
+ */
+class ToolsUiJnlpExtensionTaskSpec extends Specification {
+    private static Project rootProject
+    
+    def setupSpec() {
+        rootProject = new ProjectBuilder().withName('root').build()
+        rootProject.with {
+            version = '1.5'
+            
+            apply plugin: 'java'
+            targetCompatibility = JavaVersion.VERSION_1_7
+            
+            repositories {
+                jcenter()
+            }
+            dependencies {
+                compile 'org.slf4j:slf4j-api:1.7.7'
+                compile 'org.objenesis:objenesis:2.2'
+                runtime 'org.hamcrest:hamcrest-core:1.3'
+                testRuntime 'org.codehaus.groovy:groovy-all:2.4.5'
```

Comment:
Specifically, this integration test. Notice how the versions match. Notice also that `'org.codehaus.groovy:groovy-all:2.4.5'` doesn't appear in the control file. and won't be written by `ToolsUiJnlpExtensionTask.Writer`. That's because, below, we have:
```
dependenciesConfig = rootProject.configurations.runtime
```
meaning "only include jars that appear in the runtime configuration". `groovy-all` doesn't appear there (it's in `testRuntime`), but the other 3 do (`runtime` extends the `compile` configuration). I was just testing that ONLY jars in the specified configuration appear in the output JNLP file.